common/src/main/java/org/conscrypt/ConscryptEngineSocket.java

```diff
@@ -400,16 +400,18 @@ public final void close() throws IOException {
             stateLock.notifyAll();
         }
 
-        // Close the underlying socket.
-        super.close();
-
-        // Close the engine.
-        engine.closeInbound();
-        engine.closeOutbound();
-
-        // Release any resources we're holding
-        if (in != null) {
-            in.release();
+        try {
+            // Close the underlying socket.
+            super.close();
+
+            // Close the engine.
+            engine.closeInbound();
+            engine.closeOutbound();
```

Comment:
Should these run in a finally as well?

Reply:
Yeah, I think so.  The engine can't actually throw anything, I don't believe, so worth ensuring it gets closed down.